views/js/runner/provider/qti.js

```diff
@@ -89,7 +89,36 @@ define([
                 'serviceController',
                 'serviceExtension'
             ]);
-            return proxyFactory('qtiServiceProxy', proxyConfig);
+            return proxyFactory('qtiServiceProxy', proxyConfig)
+                // middle ware invoked on every requests
+                .use(function (req, res, next) {
```

Comment:
could you give a name to your middlewre function, this will facilitate debugging (especially if we are going to add new ones)